test/settings_test.dart

```diff
@@ -136,4 +150,200 @@ void main() {
       expect(find.text('Auto'), findsOneWidget);
     });
   });
+
+  group('Request permissions', () {
```

Comment:
Move this inside the Settings group so you don't have to copy the entire setup (just add the user mock)